src/components/Breadcrumb/Breadcrumb.jsx

```diff
@@ -0,0 +1,120 @@
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import {
+  Breadcrumb as CarbonBreadcrumb,
+  OverflowMenu as CarbonOverflowMenu,
+} from 'carbon-components-react';
+import useResizeObserver from 'use-resize-observer';
+
+import BreadcrumbOverflowItem from './BreadcrumbOverflowItem';
+import './_breadcrumb.scss';
+
+const propTypes = {
+  /** Pass in the BreadcrumbItem's for your Breadcrumb */
+  children: PropTypes.node,
+};
+
+const defaultProps = {
+  children: null,
+};
+
+const Breadcrumb = props => {
+  const { children, ...other } = props;
+  const [breadcrumbItemsWidth, setBreadcrumbItemsWidth] = useState([0]);
+  const [totalWidth, setTotalWidth] = useState(0);
+  const { ref: breadcrumbRef, width: containerWidth } = useResizeObserver({
+    useDefaults: false,
+  });
+  const [breadcrumbChildren, setBreadcrumbChildren] = useState(children);
+  const [overflowIndex, setOverflowIndex] = useState(0);
+
+  /** intial to get children total width */
+  useEffect(() => {
+    // measure child total width
+    const itemsWidth = [];
+    const breadcrumb = breadcrumbRef.current;
+    const breadcrumItems = breadcrumb.querySelectorAll('li.bx--breadcrumb-item');
```

Comment:
Instead of querying the DOM, breadcrumb items should be gathered for measurement via refs. You'll need to wrap each Carbon BreadcrumbItem in a div, and set the ref on the div since BreadcrumbItem does not utilize [forwardRef](https://reactjs.org/docs/forwarding-refs.html).